frontend/src/stimulus/controllers/dynamic/project.controller.ts

```diff
@@ -0,0 +1,342 @@
+  }
+
+  private readonly noValueOperators = ['*', '!*', 't', 'w'];
+  private readonly daysOperators = ['>t-', '<t-', 't-', '<t+', '>t+', 't+'];
+  private readonly onDateOperator = '=d';
+  private readonly betweenDatesOperator = '<>d';
+
+  setValueVisibility({ target, params: { filterName } }:{ target:HTMLSelectElement, params:{ filterName:string } }) {
+    const selectedOperator = target.value;
+    const valueContainer = this.filterValueContainerTargets.find((filterValueContainer) => filterValueContainer.getAttribute('data-filter-name') === filterName);
+    if (valueContainer) {
+      if (this.noValueOperators.includes(selectedOperator)) {
+        valueContainer.classList.add('hidden');
+      } else {
+        valueContainer.classList.remove('hidden');
+      }
+
+      if (this.daysOperators.includes(selectedOperator)) {
+        valueContainer.classList.add('days');
+        valueContainer.classList.remove('on-date');
+        valueContainer.classList.remove('between-dates');
+      } else if (selectedOperator === this.onDateOperator) {
+        valueContainer.classList.add('on-date');
+        valueContainer.classList.remove('days');
+        valueContainer.classList.remove('between-dates');
+      } else if (selectedOperator === this.betweenDatesOperator) {
+        valueContainer.classList.add('between-dates');
+        valueContainer.classList.remove('days');
+        valueContainer.classList.remove('on-date');
+      }
+    }
+  }
+
+  toggleDescription({ target, params: { projectId } }:{ target:HTMLAnchorElement, params:{ projectId:number } }) {
+    const toggledTrigger = target;
+    const otherTrigger = this.descriptionToggleTargets.find((trigger) => trigger !== toggledTrigger);
+    // The projectId action parameter is automatically typecast to Number
+    // and to compare it with a data attribute it needs to be converted to
+    // a string.
+    const clickedProjectRow = this.projectRowTargets.find((projectRow) => projectRow.getAttribute('data-project-id') === projectId.toString());
+    const projectDescriptionRow = this.descriptionRowTargets.find((descriptionRow) => descriptionRow.getAttribute('data-project-id') === projectId.toString());
+
+    if (clickedProjectRow && projectDescriptionRow) {
+      clickedProjectRow.classList.toggle('-no-highlighting');
+      clickedProjectRow.classList.toggle('-expanded');
+      projectDescriptionRow.classList.toggle('-expanded');
+
+      this.setAriaLive(projectDescriptionRow);
+    }
+
+    if (otherTrigger) {
+      otherTrigger.focus();
+    }
+  }
+
+  private setAriaLive(descriptionRow:HTMLElement) {
+    if (descriptionRow.classList.contains('-expanded')) {
+      descriptionRow.setAttribute('aria-live', 'polite');
+    } else {
+      descriptionRow.removeAttribute('aria-live');
+    }
+  }
+
+  sendForm() {
+    const ajaxIndicator = document.querySelector('#ajax-indicator') as HTMLElement;
+    ajaxIndicator.style.display = '';
```

Comment:
@oliverguenther I tried making this a target but it'd need to be a DOM descendant of the `ProjectController` element.  I tried making the `ajaxIndicator` a controller of its own and using an outlet to be able to reference it but I got:
```
setup.ts:19 Error: Missing outlet element "ajaxIndicator" for "project" controller
```

I made sure to follow the stimulus guides and add a class selector to the outlet element and upon looking up the above error, I got answers relating to outlets also needing to be within the host controller's DOM descendants and to use event emitting instead but I think this is what outlets are trying to solve so I might just have done it wrong. I'd appreciate help with this unless it's overkill for just showing the `ajaxIndicator`.

Reply:
Hi @aaron-contreras yes I think that's the right call. We might in the long run have that in its own controller (or replace it with the loading indicator anyway) so being pragmatic here is fine.